frontend/src/libs/components/shared-article-route/shared-article-route.tsx

```diff
@@ -0,0 +1,60 @@
+import { Loader, Navigate } from '~/libs/components/components.js';
+import { AppRoute, DataStatus } from '~/libs/enums/enums.js';
+import {
+  useAppDispatch,
+  useAppSelector,
+  useEffect,
+  useParams,
+} from '~/libs/hooks/hooks.js';
+import { type ValueOf } from '~/libs/types/types.js';
+import { actions as articlesActions } from '~/slices/articles/articles.js';
+
+type Properties = {
+  children: React.ReactNode;
+  redirectPath?: ValueOf<typeof AppRoute>;
+};
+
+const SharedArticleRoute: React.FC<Properties> = ({
+  children,
+  redirectPath = AppRoute.ARTICLES,
+}) => {
+  const dispatch = useAppDispatch();
+  const { user, articleId, articleIdByTokenDataStatus } = useAppSelector(
+    ({ auth, articles: { articleIdByToken, articleIdByTokenDataStatus } }) => ({
+      user: auth.user,
+      articleId: articleIdByToken,
+      articleIdByTokenDataStatus: articleIdByTokenDataStatus,
+    }),
+  );
+  const { token } = useParams();
+  const hasUser = Boolean(user);
```

Comment:
Could we just use the user in the next condition without creating a new variable?

Reply:
I left it as it is done in ProtectedRoute and PublicRoute. If I had chosen, I wouldn't have created it